applications/MeshMovingApplication/custom_processes/impose_mesh_motion_process.cpp

```diff
@@ -191,13 +203,17 @@ void ImposeMeshMotionProcess::ExecuteInitializeSolutionStep()
 {
     KRATOS_TRY;
 
-    block_for_each(mrModelPart.Nodes(),
-        [this](Node<3>& rNode) {
-            array_1d<double,3> transformed_point = rNode;
-            this->Transform(transformed_point);
-            rNode.SetValue(MESH_DISPLACEMENT, transformed_point - rNode);
-        }
-    );
+    const double time = mrModelPart.GetProcessInfo().GetValue(TIME);
+
+    if (mInterval[0] <= time && time <= mInterval[1]) {
+        block_for_each(mrModelPart.Nodes(),
+            [this](Node<3>& rNode) {
+                array_1d<double,3> transformed_point = rNode;
+                this->Transform(transformed_point);
+                rNode.GetSolutionStepValue(MESH_DISPLACEMENT) = transformed_point - rNode;
```

Comment:
```suggestion
                noalias(rNode.FastGetSolutionStepValue(MESH_DISPLACEMENT)) = transformed_point - rNode;
```

Reply:
I've never used ```noalias``` before. Can you please explain why it's necessary, or point me to some good resources?

Reply:
AFAIK there is no good docs around for `noalias`
I have sth but I only got it from colleagues

~~~
noalias is part of boost::numeric::ublas inner optimization when it's known that the right hand side doesn't share the same storage with left hand side.

 
From boost documentation:
If you know for sure that the left hand expression and the right hand expression have no common storage, then assignment has no aliasing. A more efficient assignment can be specified in this case: noalias(C) = prod(A, B); This avoids the creation of a temporary matrix that is required in a normal assignment. 'noalias' assignment requires that the left and right hand side be size conformant.

----------------------------------------------------------------------------------------
From kratos documentations:

 
In Kratos many mathematical operations and operations with variable use the library UBLAS.
for exemple, if i want to this operation or another operation:
A[a,b]=B[c][d].

X+=Y.

where A an B are matrices and a, b, c d are integers. The word "noalias" is necesary because make the operations faster. you should use it in all matrix operations. For example
noalias(A) = B.
noalias(X) += Y.
noalias(Y) = prod(A,x).
noalias(Y) = prod(trans(A),x).

prod: Matrix Products
trans:Transposed a matrix
my_matrix.resize: set the matrix with a new dimension


Other operation that we can do is to size a matrix. when we put " false" we are setting all values of the matrix o vector to be zero. The best way is for exemple:

 if( A.size1()!=3 ||  A.size2()!=3)
   A.resize( 3,3,false)


where A can be a matrix or a vector.
We can initialize with a null matrix or vector
for exemple
noalias(A)=ZeroVector(3)
noalias(A)=ZeroMatrix(4,4)

Some operations with vectors
Let see the follow exemple:

    include <boost/numeric/ublas/vector.hpp>

int main () {

   using namespace boost::numeric::ublas;
   vector<double> v1 (3), v2(3) ;
   for (unsigned i = 0; i < v.size (); ++ i)
       v1 (i) = i;
       v2 (i) = i+1;

   std::cout << sum (v) << std::endl;
   std::cout << norm_1 (v) << std::endl;
   std::cout << norm_2 (v) << std::endl;
   std::cout <<inner_prod (v1, v2)<<std::endl;
   std::cout <<outer_prod (v1, v2)<<std::endl;


}
sum: It is the sum of each component of the vector norm1: It is the root of sum of each component of the vector inner_prod : scalar product outer_product: tensor product. Only for vectors
Remember put the world Noalias in any operations

~~~

Reply:
we should probably add some documentation for it

Reply:
I see, I remember something similar with Eigen's matrix multiplication. Thanks for the docs!